File: javascript/js_002_computed_property/computed_property.js
// Title:
// Stop adding properties manually! 🛑
// Use Computed Property Names.
function getKey(key) {
  return `key is ${key}`;
}

// ❌ Bad
const user1 = {
  id: 1,
  name: "Ebrahim",
};
user1[getKey("isEnabled")] = true;

// ✅ Good
const user2 = {
  id: 1,
  name: "Ebrahim",
  [getKey("isEnabled")]: true,
};

console.log(JSON.stringify(user1) == JSON.stringify(user2));

File: python/py_ts_clips/py_004_computed_property/computed_property.py
#  Title:
#  Stop adding properties manually! 🛑
#  Use Computed Property Names.
def get_key(name):
    return f"key is {name}"

#  ❌ Bad
user1={
    "id":1,
    "name":"Ebrahim"
}
user1[get_key("is_enabled")] = True

#  ✅ Good
user2={
    "id":1,
    "name":"Ebrahim",
    get_key("is_enabled"): True
}

print(user1==user2)
